app/app/controls/BaseControl.js

```diff
@@ -235,4 +236,33 @@ export class BaseControl {
     isDragging() {
         return this.isDragging;
     }
+
+    /**
+     * Check if control is within the boundaries of given rectangle.
+     * @param x1
+     * @param y1
+     * @param x2
+     * @param y2
+     * @returns {boolean}
+     */
+    isWithinArea(x1, y1, x2, y2) {
+        var x = this.componentBodyShape.getBBox().x;
```

Comment:
Sorry for one more comment. In BaseControl is not clear that we have this.componentBodyShape variable. Developer may name it differrently in subclass. Also depending on component the approach might change. My suggestion is to make this method empty (as we do not have abstract methods in JS) and implement them for every component.